From f449c0c381239dc0f78b0f01ecfb3889e056a597 Mon Sep 17 00:00:00 2001
From: Masanori Ogino <ogino.masanori@sharp.co.jp>
Date: Tue, 22 Jan 2013 18:10:52 +0900
Subject: [PATCH 263/283] fix hover events consume issue

View#onHoverEvent() would always consume hover events over the view
if an application window has the clickable view/widget on it.
That's happened even if accessibility/talkback is disabled.
So those events will not dispatch to activity#onGenericMotionEvent().

The onHoverEvent method should return a real consumed state.

Change-Id: I9cac13b82866e5cdda0b03befb0de752a0a2e741
---
 core/java/android/view/View.java | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/core/java/android/view/View.java b/core/java/android/view/View.java
index ff44475..b3f9b31 100644
--- a/core/java/android/view/View.java
+++ b/core/java/android/view/View.java
@@ -8064,8 +8064,7 @@ public class View implements Drawable.Callback, KeyEvent.Callback,
             // in onHoverEvent.
             // Note that onGenericMotionEvent will be called by default when
             // onHoverEvent returns false (refer to dispatchGenericMotionEvent).
-            dispatchGenericMotionEventInternal(event);
-            return true;
+            return dispatchGenericMotionEventInternal(event);
         }
 
         return false;
-- 
1.8.1.1

